tests/Unit/Controller/BannerSelectionControllerTest.php

```diff
@@ -4,104 +4,50 @@
 
 namespace WMDE\BannerServer\Tests\Unit\Controller;
 
-use Symfony\Component\Asset\Packages;
 use Symfony\Component\HttpFoundation\RedirectResponse;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use WMDE\BannerServer\Controller\BannerSelectionController;
-use WMDE\BannerServer\Entity\BannerSelection\Banner;
-use WMDE\BannerServer\Entity\BannerSelection\Bucket;
-use WMDE\BannerServer\Entity\BannerSelection\Campaign;
-use WMDE\BannerServer\Entity\BannerSelection\CampaignCollection;
 use WMDE\BannerServer\Entity\BannerSelection\ImpressionThreshold;
+use WMDE\BannerServer\Tests\Fixtures\CampaignFixture;
+use WMDE\BannerServer\Tests\Fixtures\VisitorFixture;
+use WMDE\BannerServer\Tests\Utils\FakeRandomIntegerGenerator;
 use WMDE\BannerServer\UseCase\BannerSelection\BannerSelectionUseCase;
-use WMDE\BannerServer\UseCase\BannerSelection\Visitor;
-use WMDE\BannerServer\Utils\RandomInteger;
 
+/**
+ * @covers \WMDE\BannerServer\Controller\BannerSelectionController
+ * Class BannerSelectionControllerTest
+ *
+ * @package WMDE\BannerServer\Tests\Unit\Controller
+ */
 class BannerSelectionControllerTest extends \PHPUnit\Framework\TestCase {
-
-	private const VISITOR_TEST_IMPRESSION_COUNT = 5;
-	private const VISITOR_TEST_BUCKET = 'test_bucket';
-	private const VISITOR_TEST_DONATION_HISTORY = false;
-
-	private function getTestbucket(): Bucket {
-		return new Bucket(
-			'test',
-			new Banner( 'TestMain' )
-		);
-	}
-
-	private function getTrueRandomTestCampaign(): Campaign {
-		return new Campaign(
-			'C18_WMDE_Test',
-			new \DateTime( '2000-01-01 00:00:00' ),
-			new \DateTime( '2099-12-31 23:59:59' ),
-			1,
-			new RandomInteger(),
-			$this->getTestbucket()
-		);
-	}
-
-	private function getTrueRandomTestCampaignCollection(): CampaignCollection {
-		return new CampaignCollection(
-			$this->getTrueRandomTestCampaign()
-		);
-	}
-
-	private function getTestRequest(): Request {
-		return new Request(
-			[],
-			[],
-			[],
-			[
-				BannerSelectionController::IMPRESSION_COUNT_COOKIE => self::VISITOR_TEST_IMPRESSION_COUNT,
-				BannerSelectionController::BUCKET_COOKIE => self::VISITOR_TEST_BUCKET,
-				BannerSelectionController::DONATED_COOKIE => self::VISITOR_TEST_DONATION_HISTORY ]
-		);
-	}
-
-	private function getTestVisitor(): Visitor {
-		return new Visitor(
-			self::VISITOR_TEST_IMPRESSION_COUNT,
-			self::VISITOR_TEST_BUCKET,
-			self::VISITOR_TEST_DONATION_HISTORY
-		);
-	}
-
-	private function getFirstTimeVisitor(): Visitor {
-		return new Visitor(
-			0,
-			null,
-			false
-		);
-	}
-
-	private function getMockAssets(): Packages {
-		$mockAssets = $this->createMock( Packages::class );
-		$mockAssets->method( 'getUrl' )->willReturnArgument( 0 );
-		return $mockAssets;
-	}
+	private const BANNER_DIRECTORY = 'banners/';
```

Comment:
By making this smth. different, e.g. `test-banners` you could have avoided this test passing accidentally.